ethclient_utils/client.go

```diff
@@ -0,0 +1,51 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package ethclient_utils
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+
+	"github.com/ava-labs/subnet-evm/ethclient"
+	"github.com/ava-labs/subnet-evm/rpc"
+)
+
+var ErrInvalidEndpoint = errors.New("invalid rpc endpoint")
+
+func DialWithConfig(ctx context.Context, endpoint string, httpHeaders, queryParams map[string]string) (ethclient.Client, error) {
+	endpoint, err := addQueryParams(endpoint, queryParams)
+	if err != nil {
+		return nil, err
+	}
+	client, err := rpc.DialOptions(ctx, endpoint, newClientOptions(httpHeaders)...)
+	if err != nil {
+		return nil, err
+	}
+	return ethclient.NewClient(client), nil
+}
+
+// addQueryParams adds the query parameters to the url
+func addQueryParams(endpoint string, queryParams map[string]string) (string, error) {
+	uri, err := url.ParseRequestURI(endpoint)
+	if err != nil {
+		return "", fmt.Errorf("%w: %v", ErrInvalidEndpoint, err)
+	}
+	values := uri.Query()
+	for key, value := range queryParams {
+		values.Add(key, value)
+	}
+	uri.RawQuery = values.Encode()
+	return uri.String(), nil
+}
+
+// newClientOptions creates a ClientOption slice from httpHeaders
+func newClientOptions(httpHeaders map[string]string) []rpc.ClientOption {
```

Comment:
Let's name this function `newClientHeaderOptions` or similar, since the returned `rpc.ClientOption` only contains headers.